src/main/java/com/odysseusinc/arachne/portal/util/SubmissionHelper.java

```diff
@@ -329,45 +329,16 @@ public void updateExtendInfo(Submission submission) {
             try {
                 final String resultsDir = contentStorageHelper.getResultFilesDir(submission);
                 final Map<String, String> cohortNames = new HashMap<>();
-                ArachneFileMeta packageFile = findAnalysisPackage(submission);
-                if (packageFile != null) {
-                    Path tmpDir = Files.createTempDirectory("incidencerate");
-                    try {
-                        File archiveFile = tmpDir.resolve("IncidenceRate.zip").toFile();
-                        try (OutputStream out = new FileOutputStream(archiveFile);
-                             InputStream fileIn = contentStorageService.getContentByFilepath(packageFile.getPath())) {
-                            IOUtils.copy(fileIn, out);
-                        }
-                        try (FileInputStream in = new FileInputStream(archiveFile);
-                             ZipInputStream zip = new ZipInputStream(in)) {
-                            ZipEntry entry = zip.getNextEntry();
-                            while (entry != null) {
-                                if (entry.getName().endsWith(STUDY_SPECIFICATION_JSON)) {
-                                    File jsonFile = tmpDir.resolve(STUDY_SPECIFICATION_JSON).toFile();
-                                    try (FileOutputStream out = new FileOutputStream(jsonFile)) {
-                                        IOUtils.copy(zip, out);
-                                    }
-                                }
-                                zip.closeEntry();
-                                entry = zip.getNextEntry();
-                            }
-                        }
-
-                        Path specFile = tmpDir.resolve(STUDY_SPECIFICATION_JSON);
-                        if (Files.exists(specFile) && Files.isRegularFile(specFile)) {
-                            JsonParser parser = new JsonParser();
-                            try (Reader json = new FileReader(specFile.toFile())) {
-                                JsonObject spec = parser.parse(json).getAsJsonObject();
-                                JsonArray targets = spec.get("targetCohorts").getAsJsonArray();
-                                cohortNames.putAll(getCohortNames(targets));
-                                JsonArray outcomes = spec.get("outcomeCohorts").getAsJsonArray();
-                                cohortNames.putAll(getCohortNames(outcomes));
-                            }
-                        }
-                    } catch (Exception e) {
-                        LOGGER.warn("Failed to parse cohort names, {}", e.getMessage());
-                    } finally {
-                        FileUtils.deleteQuietly(tmpDir.toFile());
+                ArachneFileMeta studySpec = findAnalysisSpec(submission);
+                if (Objects.nonNull(studySpec)) {
```

Comment:
Objects.nonNull() method only exists for convenience when method reference is required, e.g. .map(Objects::nonNull)
There is no reason to use it in place of normal null check.

Reply:
skipped due to the changes from the next comment